PlayList.java

```diff
@@ -35,33 +35,62 @@ public Track getTrack(int index) {
      *  If the list is full, does nothing and returns false.
      *  Otherwise, appends the track and returns true. */
     public boolean add(Track track) {
-        //// replace the following statement with your code
-        return true;
+        boolean isAdded=false;
+        if(maxSize!=size)
+        {
+            tracks[size]=track; 
+            isAdded=true;
+            size++; //I added a track so I made the array size larger by 1.           
+        }
+        return isAdded;
     }
 
     /** Returns the data of this list, as a string. Each track appears in a separate line. */
     //// For an efficient implementation, use StringBuilder.
     public String toString() {
-        //// replace the following statement with your code
-        return "";
+        StringBuilder str=new StringBuilder();
+        for(int i=0;i<size;i++)
+        {
+            str.append(tracks[i].toString());
+            str.append("\n"); //not sure it works
+        }
+        
+        return str.toString();
     }
 
     /** Removes the last track from this list. If the list is empty, does nothing. */
      public void removeLast() {
-        //// replace this comment with your code
+        if(tracks!=null)
+        {
+            tracks[size-1]=null; //not sure if its ok
+            size--;
+        }
     }
     
     /** Returns the total duration (in seconds) of all the tracks in this list.*/
     public int totalDuration() {
-        //// replace the following statement with your code
-        return 0;
+        int total=0;
+        for(int i=0;i<size;i++)
+        {
+            total+=tracks[i].getDuration();
+        }
+        return total;
     }
 
     /** Returns the index of the track with the given title in this list.
      *  If such a track is not found, returns -1. */
     public int indexOf(String title) {
-        //// replace the following statement with your code
-        return -1;
+        String newTitle=title.toLowerCase();
+        Character.toUpperCase(newTitle.charAt(0));
+        int index=-1;
+        for(int i=0;i<size;i++)
+        {
+            if(tracks[i].getTitle()==newTitle)
+            {
+                index=i;
+            }
+        }
+        return index;
```

Comment:
1. You did manipulation over the title that you were not suppose to do.
2. You should use `title.equals` here.
3. When the index found, just return it, you don't have to iterate anymore - otherwise you'll return the lastIndexOf, while indexOf should return the first occurrence.
```suggestion
        for(int i=0;i<size;i++)
        {
            if(title.equals(tracks[i].getTitle()))
            {
                return i;
            }
        }
        return -1;
```